shell lint, better printf, colors

milljm opened the pull request:
Linted the scripts involved.

Added bold syntax highlighting for all `commands` printed.

A bit more intelligent printing of errors.

Print a final pass/fail message at the end (maybe now RED messages won't be ignored?)

Closes #28293

Commit:
shell lint, better printf, colors

Linted the scripts involved.

Added bold syntax highlighting to all `commands` printed.

A bit more intelligent printing of errors.

Print a final pass/fail message at the end (maybe now RED messages
won't be ignored?)

Closes #28293

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2373974/) on 4e9d1e7 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/28327/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2387969/) on ac5d790 wanted to post the following:


## Framework coverage

Inconsistent report tags were found between the head and base reports.
This can happen when reports are missing from either the head or the base.

Inconsistent tags:
  framework-torch-gpu
[Full coverage report](https://mooseframework.inl.gov/docs/PRs/28327/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/28327/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented on scripts/functions/diagnostic_environment.sh:
```diff
@@ -136,59 +184,80 @@ function conda_test()
-            printf "\nYour MOOSE repository requires moose-${package}:\t$(print_green ${needs_version})
-while your Conda environment has moose-${package}:\t$(print_red ${my_version})
+        if [[ "${package}" == 'none' ]]; then
+            printf "\nConda MOOSE packages not applicative.
```
what do you mean by not applicative? I dont get it

milljm replied:
I think this `if` block was meant to reside outside of the parent `if` block. Looks like a mistake to me. I'll correct it (remove it, as I see I am doing what I want a little later with an `else`.).

milljm replied:
Oh wait, no... it works like this:

```bash
moose_packages=(dev wasp libmesh petsc none)
for package in moose_packages; do
   do stuff for package
done
```
if we don't find anything and we reach a literal 'none', that means the user has moose-mpi installed. And it is up to them to build petsc, libmesh, wasp manually.

milljm replied:
example:
```bash
conda create -n testing moose-mpi packaging jinja2 pyaml
conda activate testing
./diagnostics
<trimmed>
###############################################################################################
Python Modules (TestHarness, run-ability)

OK
###############################################################################################
CONDA MOOSE Package Checks


Conda MOOSE packages not applicative.
User is required to manually build PETSc, libMesh, and WASP


checks PASSED
```

GiudGiud replied:
`Conda MOOSE packages not applicative.`
what do you mean by this? 
Maybe "conda installation path is not selected"

milljm replied:
But you are:

```bash
conda create -n testing moose-mpi
                        ^^^^^^^^^
```
This is basically supporting all you devs, that refuse to install `moose-petsc` or `moose-libmesh`, but you want the compiler stack.

milljm replied:
How about this: you tell me what you want it to say?

Like:
```pre
Conda MOOSE support library packages not installed.
```

GiudGiud replied:
yeah basically. That s how I understand it. 
If it s correct then works for me

milljm replied:
How about this, since if we hit 'none' we no we're missing all of them:

```pre
Conda MOOSE dependencies not avilable (moose-petsc, moose-libmesh,
moose-wasp, etc). User is required to manually build PETSc, libMesh,
and WASP.
```

milljm commented on scripts/diagnostics.sh:
```diff
@@ -21,98 +27,151 @@ function print_failure_and_exit()
 
 function print_help()
 {
+    # shellcheck disable=SC2034
```
These are shell lint controls (like pylint). It is the official way to document shell scripts.

milljm replied:
This is how they render an IDE with shellcheck enabled:

\# shellcheck disable=[SC2034](https://www.shellcheck.net/wiki/SC2034)

Commit:
Additional wordage when using specific packages

Add more information upon hitting the logic specifying when a user
must build their own support packages.

GiudGiud commented on scripts/functions/diagnostic_environment.sh:
```diff
@@ -216,8 +217,9 @@ function conda_test()
         if [[ "${package}" == 'none' ]]; then
-            printf "\nConda MOOSE packages not applicative.
-User is required to manually build PETSc, libMesh, and WASP\n\n"
+            printf "\nConda MOOSE dependencies not avilable (moose-petsc, moose-libmesh,
```
available

GiudGiud commented on scripts/functions/diagnostic_environment.sh:
```diff
@@ -216,8 +217,9 @@ function conda_test()
-User is required to manually build PETSc, libMesh, and WASP\n\n"
+            printf "\nConda MOOSE dependencies not avilable (moose-petsc, moose-libmesh,
+moose-wasp, etc). User is required to manually build PETSc, libMesh,
+and WASP\n\n"
```
and WASP (see moose/scripts folder)

GiudGiud reviewed (approved):
some suggestions
works for me

Commit:
spelling/grammar, addition

milljm commented:
After discussing, there is an issue I still need to address.

Commit:
List all package information

List all information possible, highlighting the issue present.
The idea is to give the user (and developer helping the user)
as much information as to what is going on as possible.

Commit:
better conda install line

Commit:
print recovery steps

add recovery steps to be printed based on circumstances.

Commit:
append `mpich` (or our default)

Append the default MPI implementation to the list of Conda packages
to install.
